Antidote/AbstractCallViewController.h

```diff
@@ -18,21 +16,20 @@
 + (instancetype)new NS_UNAVAILABLE;
 
 /**
- * Create an instance of the CallViewController.
- * @param chat Appropriate chat for the call.
- * @param manager The call maanger for the application.
+ * Create an instance of the AbstractCallViewController.
+ * @param nickname The nickname associated with the caller.
  */
-- (instancetype)initWithCall:(OCTCall *)call submanagerCalls:(OCTSubmanagerCalls *)manager;
+- (instancetype)initWithCallerNickname:(NSString *)nickname;
 
 /**
- * The call associated with this view controller.
+ * Abstract delegate
  */
-@property (strong, nonatomic, readonly) OCTCall *call;
+@property (weak, nonatomic) id delegate;
```

Comment:
We can remove it but then it would make it less convenient for this method below in Calls Manager
```
- (AbstractCallViewController *)viewControllerForCall:(OCTCall *)call
{
    AbstractCallViewController *viewController;

    OCTFriend *friend = [call.chat.friends firstObject];

    switch (call.status) {
        case OCTCallStatusActive:
            viewController = [[ActiveCallViewController alloc] initWithCallerNickname:friend.nickname];
            break;
        case OCTCallStatusDialing:
            viewController = [[DialingCallViewController alloc] initWithCallerNickname:friend.nickname];
            break;
        case OCTCallStatusRinging:
            viewController = [[RingingCallViewController alloc] initWithCallerNickname:friend.nickname];
            break;
        case OCTCallStatusPaused:
            NSAssert(NO, @"We should not be here yet. Not yet implemented");
            break;
    }

    viewController.modalInPopover = YES;
    viewController.modalPresentationStyle = UIModalPresentationOverCurrentContext;
    viewController.delegate = self;

    return viewController;
}
```

Without the delegate in `Abstract` I would not be able to assign a delegate as easily. Which is why I use `dynamic` in the other subclasses. Perhaps there's a better way?